packages/acs-ui-common/src/common.ts

```diff
@@ -45,3 +45,81 @@ export const _preventDismissOnEvent = (
 export function _getKeys<T extends object>(obj: T): (keyof T)[] {
   return Object.keys(obj) as Array<keyof T>;
 }
+
+/**
+ * Data model that represents a chat message attachment
+ * where it contains an ID to uniquely identify the attachment,
+ * a name that represents the name of file, and
+ * a URL to download the attachment.
+ *
+ * @beta
+ */
+export interface AttachmentMetadata {
+  /**
+   * Unique ID of the attachment.
+   */
+  id: string;
+  /**
+   * Attachment name to be displayed.
+   */
+  name: string;
+  /**
+   * Download URL for the attachment.
+   */
+  url: string;
+}
+
+/**
+ * Data model that represents a chat message attachment being uploaded
+ * where it contains an ID to uniquely identify the attachment,
+ * a name that represents the name of file,
+ * an optional URL to download the attachment,
+ * an optional progress value between 0 and 1 indicating the progress of the upload, and
+ * an optional error object that contains error message would be shown to the user.
+ *
+ *
+ * @beta
+ */
+export interface AttachmentMetadataWithProgress {
```

Comment:
Should we rename this to something like `AttachmentInProgressMetadata`?

Reply:
thanks updated